crates/evm/src/query.rs

```diff
@@ -735,7 +735,17 @@ impl<C: sov_modules_api::Context> Evm<C> {
         working_set: &mut WorkingSet<C>,
     ) -> RpcResult<reth_primitives::U256> {
         let estimated = self.estimate_tx_expenses(request, block_number, working_set)?;
-        Ok(estimated.gas_with_l1_overhead())
+
+        // TODO: this assumes all blocks have the same gas limit
+        // if gas limit ever changes this should be updated
+        let last_block = self
```

Comment:
Why is it last_block, not a block by block_number?

Reply:
all blocks have the same gas limit on citrea.

until we do a fork and increase or decrease the gas limit, it doesnt matter which block we use there.

but once we do, we have to change that part, hence the todo